### PR TITLE
electron frontend for fidor payments

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: src/main/electron/index.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<h1>
```
Comment (nerdsupremacist) [inline src/main/electron/index.html]: Do we need this file?

Comment (nerdsupremacist) [reply src/main/electron/index.html]: Shame 🔔  Shame 🔔 Shame 🔔

File: src/main/electron/index.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+} else {
+    serverProcess = childProcess
+        .spawn('java', ['-jar', app.getAppPath() + appPath], {env: env});
+
+    serverProcess.stdout.on('data', function (data) {
+        console.log(data.toString());
+    });
+
+    serverProcess.stderr.on('data', function (data) {
+        console.log(data.toString());
```
Comment (nerdsupremacist) [inline src/main/electron/index.js]: Maybe `console.error` this part

File: src/main/electron/index.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    });
+}
+
+let appUrl = 'http://localhost:1337';
```
Comment (nerdsupremacist) [inline src/main/electron/index.js]: Nice 👍

File: src/main/scala/org/breakout/connector/fidor/FidorOAuthServer.scala [hunk 4]
```diff
@@ -14,33 +20,50 @@ object FidorOAuthServer {
...
 
       apiCodeOption match {
         case Some(_) =>
-          Ok().withBody("all ok, this can be closed")
+          usageEnvironment match {
+            case WEB_FRONTEND =>
+              Ok("<script>window.location.href = '/transactions';</script>")
+                .putHeaders(Header("Content-Type", "text/html; charset=UTF-8"))
+            case CLI =>
+              Ok("<script>window.close();</script>")
```
Comment (nerdsupremacist) [inline src/main/scala/org/breakout/connector/fidor/FidorOAuthServer.scala:51]: huh? why are you responding with javascript? When used by the CLI?

File: src/main/scala/org/breakout/http/Frontend.scala [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+          backendPayments.exists(_.fidorId == transaction.id.toLong)
+        }
+
+        Future.sequence(withoutAlreadyTransferred.map { transaction =>
+          BackendApi.addPayment(
+            transaction.subject.getSubjectCode,
+            BackendPayment(transaction.amount.toDecimalAmount, transaction.id.toLong, transaction.booking_date.flatMap(_.toUtcLong))
+          ) map { invoice =>
+            log.info(s"SUCCESS: inserted payment to backend invoice $invoice")
+            Right(transaction, invoice)
```
Comment (nerdsupremacist) [inline src/main/scala/org/breakout/http/Frontend.scala:53]: This appears to be some enum. Are you just separating between successful and failed transactions?

**Review by nerdsupremacist — CHANGES_REQUESTED** [review]

File: src/main/scala/org/breakout/connector/fidor/FidorOAuthServer.scala (hunk 4 above)
Comment (nerdsupremacist) [reply src/main/scala/org/breakout/connector/fidor/FidorOAuthServer.scala:51]: never mind. I got it

File: src/main/scala/org/breakout/http/Frontend.scala (hunk 5 above)
Comment (thepiwo) [reply src/main/scala/org/breakout/http/Frontend.scala:53]: exactly

**Review by nerdsupremacist — APPROVED** [review]
lgtm

**Pull request merged**